MsFinderTest/MsfinderConsoleApp.cs

```diff
@@ -0,0 +1,180 @@
+            {
+                FileStorageUtility.DeleteSfdFiles(formulaFile);
+            }
+
+            foreach (var rawData in _rawDataList)
+            {
+                var formulaResult = MolecularFormulaFinder.GetMolecularFormulaScore(_productIonDB, _neutralLossDB, _existFormulaDB, rawData, _param);
+                var formualResults = new List<FormulaResult>() { formulaResult };
+                FormulaResultParcer.FormulaResultsWriter($"{_projectDir}/testdata/input/test.fgt", formualResults);
+            }
+
+            var output = $"{_projectDir}/testdata/input/test.fgt";
+            var expected = $"{_projectDir}/testdata/expected/test_expected.fgt";
+
+            var out_hash = GetHashCode(output);
+            var expected_hash = GetHashCode(expected);
+
+            Assert.Equal(out_hash, expected_hash);
+
+        }
+
+        [Fact]
+        public void FragmenterResult()
+        {
+            var structureFile = System.IO.Directory.GetFiles($"{_projectDir}/testdata/input/test", "test.sfd");
+            if (structureFile.Length > 0)
+            {
+                FileStorageUtility.DeleteSfdFiles(structureFile);
+            }
+
+            foreach (var rawData in _rawDataList)
+            {
+                var formulaResult = MolecularFormulaFinder.GetMolecularFormulaScore(_productIonDB, _neutralLossDB, _existFormulaDB, rawData, _param);
+                if (rawData.Ms2PeakNumber <= 0 || rawData.Smiles == null || rawData.Smiles == string.Empty) return;
+                var structureQuery = new ExistStructureQuery(rawData.Name, rawData.InchiKey, rawData.InchiKey, new List<int>(), formulaResult.Formula, rawData.Smiles, string.Empty, 0, new DatabaseQuery());
+                if (structureQuery == null) return;
+
+                var eQueries = new List<ExistStructureQuery>() { structureQuery };
+
+                var adductIon = AdductIonParcer.GetAdductIonBean(rawData.PrecursorType);
+                var centroidSpectrum = FragmentAssigner.GetCentroidMsMsSpectrum(rawData);
+                var refinedPeaklist = FragmentAssigner.GetRefinedPeaklist(centroidSpectrum, _param.RelativeAbundanceCutOff,
+                    rawData.PrecursorMz, _param.Mass2Tolerance, _param.MassTolType, true);
+                var curatedPeaklist = PeakAssigner.getCuratedPeaklist(formulaResult.ProductIonResult);
+
+                var results = Riken.Metabolomics.StructureFinder.MainProcess.Fragmenter(eQueries,
+                                rawData, curatedPeaklist, refinedPeaklist, adductIon, formulaResult,
+                                _param, null, _fragmentOntologyDB);
+
+                foreach (var result in results)
+                {
+                    result.TotalScore += formulaResult.TotalScore;
+                    result.Ontology = rawData.Ontology;
+                }
+
+                FragmenterResultParcer.FragmenterResultWriter($"{_projectDir}/testdata/input/test/test.sfd", results, true);
```

Comment:
If these functions are exactly the same as the ones called in the functions from the package, can we either extract a function in the package or call the package function directly?

Reply:
updated with the function from the package in [dec9dec](https://github.com/RECETOX/recetox-msfinder/pull/22/commits/dec9decb3b591ff6904b28fe21abb6d3c956b48f)